src/test/java/com/example/LongestPartStartIndexFinderTest.java

```diff
@@ -0,0 +1,34 @@
+package com.example;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static com.example.LongestPartStartIndexFinder.findIndexOfLongestPart;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class LongestPartStartIndexFinderTest {
+
+    static Stream<Arguments> testData() {
+        return Stream.of(
+                Arguments.of("aaaaabbbcccaaaaaa", 11),
+                Arguments.of("aaabbbccaabaaaaac", 11),
+                Arguments.of("aaabbbbbccddeeffedd", 3),
+                Arguments.of("aaabbbccccddd", 6),
+                Arguments.of("abbbcccdddeee", 1),
+                Arguments.of("aaabbbcccddd", 0),
+                Arguments.of("abcdef", 0),
+                Arguments.of("aa", 0),
+                Arguments.of("", 0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("testData")
+    void findIndex(String input, int expectedIndex) {
+        var calculatedIndex = input.chars().boxed().collect(findIndexOfLongestPart());
```

Comment:
Would be much simpler just to call the static method for each input.
There is risk here that test code can contain bug as well.
One related article: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

Reply:
In this case I think the parameterized test like same you mentioned. Because parameterized test calls test method each test data, like as you write inputs line by line. The test method doesn't create input and expected data, only runs defined test collection.